pkg/controller/job/job_controller_test.go

```diff
@@ -962,6 +962,9 @@ func checkIndexedJobPods(t *testing.T, control *controller.FakePodControl, wantI
 	gotIndexes := sets.New[int]()
 	for _, p := range control.Templates {
 		checkJobCompletionEnvVariable(t, &p.Spec)
+		if feature.DefaultFeatureGate.Enabled(features.PodIndexLabel) {
```

Comment:
The pattern is usually to explicity enable the unit test as a parameter.

I would pass in this feature toggle as a test case parameter and verify the negative and positive case.  

I know this is default enabled but you still want to test that this works if an admin turned this off.

Reply:
Implemented your suggestion, let me know how it looks.

Since the feature flag is default enabled, i opted to invert the usual parameter pattern `<featureflag>Enabled bool` flag to `<featureflag>Disabled bool`, so that all the existing test cases that don't specify this parameter will execute with the feature flag enabled, and then I added one test case which sets this parameter to true to execute indexed job test case with the feature flag disabled. This way we achieve full unit test coverage for a "default enabled" feature which I think is important. 

What do you think? Does this approach make sense or is it there a better alternative?

Reply:
Just realized I forgot to push the changes described in my previous comment 😅 Just pushed the changes. Let me know what you think.

Reply:
TBH I don't really understand how feature flags work in k/k but I don't see a lot of test code that uses this pattern.  

I think you should just check the flag you pass in rather than the feature toggle as this seems different from the rest of the code.

```
if !tc.podIndexLabelDisabled {...}
```